### PR TITLE
Add method to delete an index without throwing any error (`DeleteIfExist`, `DeleteIndexIfExist`)

### LINKED ISSUE #156: Add a method to delete the index only if it already exists
Trying to delete an index that does not exist on the MeiliSearch server will throw an error. To delete it without any error no matter what if it exists or not, we need a new method.

- [ ] Add a method `delete_index_if_exists`: it deletes the index but does not throw any error if the index does not exist.
- [ ] Add tests

Example: https://github.com/meilisearch/meilisearch-python/pull/268/files

⚠️ This issue is generated, and `delete_index_if_exists` might be named differently in this package. Keep the already existing way of naming in this package to stay idomatic with the language and the repository.

Related to https://github.com/meilisearch/integration-guides/issues/107

Sorry if this is already partially/completely implemented, feel free to let me know about the state of this issue.

### PR DESCRIPTION
**Pull request by jdkramhoft**
- Added DeleteIfExist method to Index
- Added DeleteIndexIfExist method to Client
- Added unit tests for existing/non-existing index

Closes #156

### CODE REVIEW

**Comment by curquiza** [comment]
bors try

File: src/Meilisearch/MeilisearchClient.cs [hunk 1]
```diff
@@ -201,5 +201,30 @@ public async Task<bool> DeleteIndex(string uid)
...
+
+        /// <summary>
+        /// Deletes the index if it exists.
+        /// It's not a recovery delete. You will also lose the documents within the index.
+        /// </summary>
+        /// <param name="uid">unique dump identifier.</param>
+        /// <returns>Returns the status of the delete operation.
+        /// True if the index existed and was deleted. False if it did not exist. </returns>
+        public async Task<bool> DeleteIndexIfExists(string uid)
+        {
```
Comment (curquiza) [inline src/Meilisearch/MeilisearchClient.cs:203]: Could you call `DeleteIfExists` in `DeleteIndexIfExists` to avoid code duplication? :)

**Review by curquiza — CHANGES_REQUESTED** [review]
Hello @jdkramhoft 
sorry for the delay and thanks for this addition!!

**Comment by curquiza** [comment]
Also, it looks like your tests fail. Please follow the CONTRIBUTING guide to know how to run the tests locally and to fix them: https://github.com/meilisearch/meilisearch-dotnet/blob/main/CONTRIBUTING.md#development-workflow

File: src/Meilisearch/Index.cs [hunk 2]
```diff
@@ -84,6 +84,36 @@ public async Task<bool> Delete()
...
+        /// Deletes the index if it exists.
+        /// It's not a recovery delete. You will also lose the documents within the index.
+        /// </summary>
+        /// <returns>Returns the status of the delete operation.
+        /// True if the index existed and was deleted. False if it did not exist. </returns>
+        public async Task<bool> DeleteIfExists()
+        {
+            try
+            {
+                var responseMessage = await this.client.DeleteAsync($"/indexes/{this.Uid}");
```
Comment (curquiza) [inline src/Meilisearch/Index.cs]:
```suggestion
                var responseMessage = await this.http.DeleteAsync($"/indexes/{this.Uid}");
```

**Comment by curquiza** [comment]
bors try

File: src/Meilisearch/MeilisearchClient.cs (hunk 1 above)
Comment (jdkramhoft) [reply src/Meilisearch/MeilisearchClient.cs:203]: Whoops :) 
I have done this now

**Comment by curquiza** [comment]
bors try

**Review by curquiza — APPROVED** [review]
bors merge

thanks a lot @jdkramhoft 


If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2021/), and you would like to receive a small gift from MeiliSearch too, please complete [this form](https://meilisearch.typeform.com/to/rPexR5gq).

**Pull request merged**